model-datastructure/src/commonTest/kotlin/LinearHistoryTest.kt

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright (c) 2023.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import org.modelix.model.LinearHistory
+import org.modelix.model.VersionMerger
+import org.modelix.model.lazy.CLTree
+import org.modelix.model.lazy.CLVersion
+import org.modelix.model.lazy.ObjectStoreCache
+import org.modelix.model.operations.IOperation
+import org.modelix.model.persistent.MapBaseStore
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class LinearHistoryTest {
+    val initialTree = CLTree.builder(ObjectStoreCache(MapBaseStore())).repositoryId("LinearHistoryTest").build()
+
+    @Test
+    fun simpleTest1() {
+        val v20 = version(20, null)
+        val v21 = version(21, null)
+
+        assertHistory(v20, v21, listOf(v20, v21))
+    }
+
+    @Test
+    fun simpleTest2() {
+        val v10 = version(10, null)
+        val v20 = version(20, v10)
+        val v21 = version(21, v10)
+
+        assertHistory(v20, v21, listOf(v20, v21))
+    }
+
+    @Test
+    fun linearHistoryPerformance() {
```

Comment:
If there was ever a bug, these random variable names and IDs would make debugging pretty annoying.

Reply:
This test is just a dump from one of the other failing tests to make debugging easier.